File: 3) Pascal's Triangle/Pascal's Triangle.py
def generate(n: int):
    pascal_list = [[1]]   # Initializing Pascal's Triangle
    # Generate subsequent rows (excluding the first row).
    for _ in range(n-1):
        # Temporary list by padding the previous row with zeros on both sides
        temp = [0] + pascal_list[-1] + [0]
        # Generating the Current Row
        row = [temp[j] + temp[j + 1] for j in range(len(pascal_list[-1]) + 1)]
        # Append the generated row to Pascal's Triangle.
        pascal_list.append(row)
    return pascal_list

n = int(input("Enter the number of rows "))
print(generate(n))

File: 3) Pascal's Triangle/Pascal's Triangle.js
function generate(n) {
    let pascalList = [[1]];
    for (let i = 0; i < n - 1; i++) {
        let temp = [0, ...pascalList[pascalList.length - 1], 0];

        let row = temp.map((val, index) => val + temp[index + 1]);

        row.pop();
        pascalList.push(row);
    }
    return pascalList;
}

let n = 6;
console.log(generate(n));
